### PR TITLE
bb-4 setup events commands structure

### PR DESCRIPTION
**Pull request by yli-yasir**

### CODE REVIEW

File: src/events/ready.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export default {
+    name: 'ready',
+    execute: ()=>{
+        console.log('Logged in!')
```
Comment (MaybeAcid) [inline src/events/ready.js:4]: Don't have functions that do nothing.

File: src/utils.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import Discord from 'discord.js';
+import path from 'path';
+import fs from 'fs';
+
+export async function getCommandsCollection() {
```
Comment (MaybeAcid) [inline src/utils.js:5]: We are using es6, so arrow functions would be preferred.

`export const getCommandsCollection = async () => {...}`

File: src/utils.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    const commandsCollection = new Discord.Collection();
+    await forEachModule('./commands', (cmdModule) => {
```
Comment (MaybeAcid) [inline src/utils.js:7]: `await forEachModule('./commands', cmdModule => {`

you  don't need brackets

File: src/utils.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    await forEachModule('./events', (eventModule) => {
+        const event = eventModule.default;
+        const { name, execute } = event;
+        discordClient.on(name, (...args) => execute(...args));
+    })
+}
+
+export function parseCommand(command) {
+    const prefix = process.env.DISCORD_BOT_PREFIX;
+    const args = command.slice(prefix.length).trim().split(/ +/);
```
Comment (MaybeAcid) [inline src/utils.js:24]: I don't know what this means.  could you rename the variables to something that makes sense? you're slice, trimming and splitting.

Could you also tell me what you your goals is? and what command returns?

File: src/utils.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    await forEachModule('./commands', (cmdModule) => {
+        const command = cmdModule.default;
+        commandsCollection.set(command.name, command)
+    })
+    return commandsCollection;
+}
+
+export async function registerEventHandlers(discordClient) {
+    await forEachModule('./events', (eventModule) => {
+        const event = eventModule.default;
```
Comment (MaybeAcid) [inline src/utils.js:16]: event isn't being used anywhere else change it to the following: 
`const { name, execute } = eventModule.default;`

File: src/index.js [hunk 6]
```diff
@@ -1,14 +1,16 @@
 import Discord from 'discord.js';
+import { getCommandsCollection, registerEventHandlers } from './utils';
+
 const client = new Discord.Client();
 
-client.on('ready', () => {
-  console.log(`Logged in as ${client.user.tag}!`);
-});
+async function start() {
```
Comment (MaybeAcid) [inline src/index.js:6]: use arrow functions.

File: package.json [hunk 7]
```diff
@@ -26,6 +26,7 @@
   "devDependencies": {
     "@babel/cli": "^7.13.0",
     "@babel/core": "^7.13.8",
+    "@babel/plugin-syntax-dynamic-import": "^7.8.3",
```
Comment (MaybeAcid) [inline package.json:29]: is this required for this pr to work?

File: src/events/ready.js (hunk 1 above)
Comment (yli-yasir) [reply src/events/ready.js]: removed

File: src/utils.js (hunk 4 above)
Comment (yli-yasir) [reply src/utils.js]: The goal is to pass a command (A message which starts with the bots prefix) , and get back the command name and arguments.

I renamed args to commandArguments. Like the name implies the Command arguments are returned as an array.

First the prefix is sliced off, which leaves you with the command name and arguments. Then trimming is done to remove any spaces on the sides. Finally the remaining string is split by one or more whitespace by utilizing a regex.

This results in an array which contains the command name as the first element and the arguments. 

The command name is then removed on line 25 and stored as a separate variable.

File: package.json (hunk 7 above)
Comment (yli-yasir) [reply package.json:29]: Haven't tested without it. According to online documentation it is required for babel to handle the dynamic import syntax.

import(modulePath)

**Review by MaybeAcid — APPROVED** [review]

**Pull request merged**